src/main/java/Duke.java

```diff
@@ -1,10 +1,162 @@
+    }
+
+    /**
+     * Greets the user by printing some introductory messages
+     */
+    private static void greet() {
+        printLine();
+        System.out.println("Hello! I'm Duke");
+        System.out.println("What can I do for you?");
+        printLine();
+    }
+
+    /**
+     * Prints a line on the screen
+     */
+    public static void printLine() {
+        System.out.println("____________________________________________________________");
+    }
+
+    /**
+     * Takes in input from the user and executes the given instructions.
+     * If user input is "list", it calls the list() function to list all the tasks.
+     * If user input is done x, where x is a valid task number, it calls the MarkTaskAsDone() function to mark the task as done.
+     * If user has scheduled a task by writing a statement beginning with "event", "deadline" or "todo",
+     * then it adds the task in the tasks list by calling the addTaskToList() function.
+     *
+     * @param userInput            userInput stores the input String entered by the user.
+     * @param taskCompletionStatus taskCompletionStatus Stores the status of the task, true if completed, false otherwise.
+     */
+    private static void runDuke() {
+        int i;
+        String userInput;
+        String taskCompletionStatus = "";
+        Scanner in = new Scanner(System.in);
+        userInput = in.nextLine();
+
+        while (!(userInput.equals("bye"))) {
+            if (userInput.equals("list")) {
+                List();
+            } else if (userInput.startsWith("done")) {
+                markTaskAsDone(userInput);
+            } else {
+                addTaskToList(userInput);
+            }
+            printLine();
+            userInput = in.nextLine();
+        }
+    }
+
+    /**
+     * Greets the user goodbye and the code finishes its execution.
+     */
+    private static void greetBye() {
+        printLine();
+        System.out.println(" Bye. Hope to see you again soon!");
+        printLine();
+    }
+
+    /**
+     * Adds the task to the task list if its a valid task creation statement given by the user.
+     * Displays appropriate message if task is valid and is created successfully, vice versa.
+     * Displays the total number of tasks in the list as well.
+     *
+     * @param taskDescription TaskDescription Stores the description of the task.
+     * @param taskDueBy       TaskDueBy stores the deadline of the task which is created as a deadline.
+     * @param taskDueAt       TaskDueAt stores the event timing of the task created as an event.
+     * @param isTaskValid     IsTaskValid stores true if the task statement emtered by the user is a valid task creation statement and fasle, otherwise.
+     * @param index           Index stores the index of the "/" in the entered String
+     */
+    private static void addTaskToList(String userInput) {
+
+        String taskDescription;
+        String taskDueBy;
+        String taskDueAt;
```

Comment:
Perhaps for events, taskDueAt may not be the most intuitive name, maybe I could suggest taskDuration? That's if it helps you to understand it better, if not it is fine too.

Reply:
I chose TaskDueAt because we are using the "at" to distinguish an event, so i somehow wanted to use "At" in the string name to make it clear for the readers. But I will definitely consider changing the names.